movies/4.create-movie.md

```diff
@@ -0,0 +1,212 @@
+          "description": "Textual description of the movie."
+        },
+        {
+          "property": "schema:duration",
+          "title": "duration",
+          "description": "The movie's duration in ISO 8601 date format."
+        }
+      ]
+    }
+  ]
+}
+```
+
+```http
+POST https://hydra-movies.herokuapp.com/movies
+Content-Type: application/ld+json
+
+{
+  "@context": "https://schema.org/",
+  "@type": "Movie",
+  "name": "Pulp Fiction",
+  "description": "Pulp Fiction is a 1994 American crime film. It tells several stories of criminal Los Angeles.",
+  "duration": "PT2H34M"
+}
+```
+
+Response (excerpt):
+
+```http
+HTTP/1.1 201 Created
+Location: /movies/d517cae6-6cdc-11e9-a923-1681be663d3e
+Content-Type: application/ld+json
+
+{
+  "@context": "https://hydra-movies.herokuapp.com/context.jsonld",
+  "@type": "schema:Movie",
+  "@id": "/movies/d517cae6-6cdc-11e9-a923-1681be663d3e",
+  "schema:name": "Pulp Fiction"
+}
+```
+
+
+### Explanation
+
+```
+"collection": {
+    "@id": "movies",
+    "@type": [
+      "Collection",
+      "mov:Movies"
+    ]
+  }
+```
+
+By loading the API entry point a client came across a resource of type `mov:Movies`. To understand what it can do with it, it consults the API documentation.
+ 
+```
+"supportedClass": [
+  {
+    "@id": "mov:Movies",
+    "@type": "Class",
+```
+
+The type can be found within `supportedClass`.
+
+```
+"supportedOperation": {
+  "@id": "mov:create-movie",
+  "@type": [
+    "Operation",
+    "schema:CreateAction"
+  ],
+  "title": "Create movie",
+  "description": "Creates a new movie",
+```
+
+The `supportedOperation` of `mov:Movies` describe what can be done on resources of that type. `Operation` is the hydra type for any API operation. By adding `schema:CreateAction` we can be more precise and define it as an operation that creates new resources. `title` and `description` allow a friendly human-readable output in generated developer documentation and generic clients.
+
+```
+"method": "POST",
+"expects": "schema:Movie",
+"returns": "schema:Movie"
+```
+
+The `method` is exactly telling the client, that it should use the HTTP POST verb to use this operation. `expects` and `returns` specify which types the operation expects as input and returns in response.
+
+```
+{
+    "@id": "schema:Movie",
+    "@type": "Class",
+    "title": "Movie",
+    "description": "A single movie",
+```
+
+This type is also present in the API documentation as a `supportedClass`. Again `title` and `description` give some info for humans.
+
+```
+"supportedProperty": [
+  {
+    "property": "schema:name",
+    "title": "name",
+    "description": "The movie's name"
+  },
+```
+
+To construct a HTTP request the client needs to know which properties are supported by the expected type `schema:Movie`. The API documentation contains this information in `supportedProperty`. The client now is aware that it can send `schema:name`, `schema:description` and `schema:duration`.
```

Comment:
```suggestion
To construct an HTTP request the client needs to know which properties are supported by the expected type `schema:Movie`. The API documentation contains this information in `supportedProperty`. The client now is aware that it can send `schema:name`, `schema:description` and `schema:duration`.
```